keps/sig-cluster-lifecycle/image-builder/20190612-image-builder.md

```diff
@@ -0,0 +1,122 @@
+---
+kep-number: 37
+title: Kubernetes Image Builder
+status: provisional
+authors:
+  - "@timothysc"
+  - "@moshloop"
+owning-sig: sig-cluster-lifecycle
+reviewers:
+  - "TBD"
+approvers:
+  - "@justinsb"
+  - "@timothysc"
+  - "@luxas"
+editor:
+  - "@moshloop"
+  - "@timothysc"
+creation-date: 2019-06-11
+last-updated: 2019-06-11
+---
+
+# Kubernetes Image Builder
+
+## Table of Contents
+
+* [Kubernetes Image Builder](#kubernetes-image-builder)
+  * [Metadata](#metadata)
+  * [Table of Contents](#table-of-contents)
+  * [Summary](#summary)
+  * [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non\-goals](#non-goals)
+  * [Proposal](#proposal)
+  * [Graduation Criteria](#graduation-criteria)
+  * [Implementation History](#implementation-history)
+  * [Alternatives](#alternatives)
+
+## Summary
+It is common for modern cloud based software deployments to follow immutable patterns. One of the foundational pieces to this idea is the creation of immutable images. There are already several tools that create images in the kubernetes ecosystem, which include: [Wardroom](https://github.com/heptiolabs/wardroom), [Cluster API AWS](https://github.com/kubernetes-sigs/cluster-api-provider-aws/tree/master/build), [Cluster API vSphere](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/tree/master/docs/ova), [amazon-eks-ami](https://github.com/awslabs/amazon-eks-ami), [talos](https://docs.talos-systems.com/), [LinuxKit](https://github.com/linuxkit/linuxkit), etc. The purpose of this proposal is to distill down the common requirements and provide the Kubernetes ecosystem with a standard image building utility that can be leveraged by the Kubernetes ecosystem.  
```

Comment:
Can we include https://github.com/kubernetes/kube-deploy/tree/master/imagebuilder ?  I think it would be great to fold that project in also.